src/components/app/details/cicdHistory/LogsRenderer.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { Progressing, useInterval } from '../../../common'
+import { useParams } from 'react-router'
+import { ReactComponent as OpenInNew } from '../../../../assets/icons/ic-open-in-new.svg'
+import { ReactComponent as Info } from '../../../../assets/icons/info-filled.svg'
+import { ReactComponent as Question } from '../../../../assets/icons/ic-help.svg'
+import { DOCUMENTATION, EVENT_STREAM_EVENTS_MAP, Host, LOGS_RETRY_COUNT, POD_STATUS, Routes } from '../../../../config'
+import { default as AnsiUp } from 'ansi_up'
+import { HistoryComponentType, LogsRendererType } from './types'
+
+export default function LogsRenderer({
+    triggerDetails,
+    isBlobStorageConfigured,
+    parentType,
+}: LogsRendererType): JSX.Element {
+    const { pipelineId, envId, appId } = useParams<{ pipelineId: string; envId: string; appId: string }>()
+    const logsURL = parentType === HistoryComponentType.CI
+    ? `${Host}/${Routes.CI_CONFIG_GET}/${pipelineId}/workflow/${triggerDetails.id}/logs`
+    : `${Host}/${Routes.CD_CONFIG}/workflow/logs/${appId}/${envId}/${pipelineId}/${triggerDetails.id}`
+    const [logs, eventSource, logsNotAvailable] = useCIEventSource(
+        triggerDetails.podStatus && triggerDetails.podStatus !== POD_STATUS.PENDING && logsURL
+    )
+    function createMarkup(log) {
+        try {
+            log = log.replace(/\[[.]*m/, (m) => '\x1B[' + m + 'm')
+            const ansi_up = new AnsiUp()
+            return { __html: ansi_up.ansi_to_html(log) }
+        } catch (err) {
+            return { __html: log }
+        }
+    }
+
+    return triggerDetails.podStatus !== POD_STATUS.PENDING &&
+        logsNotAvailable &&
+        (!isBlobStorageConfigured || !triggerDetails.blobStorageEnabled) ? (
+        renderConfigurationError(isBlobStorageConfigured)
+    ) : (
+        <div className="logs__body">
+            {logs.map((log, index) => {
+                return <p className="mono fs-14" key={`logs-${index}`} dangerouslySetInnerHTML={createMarkup(log)} />
+            })}
+            {(triggerDetails.podStatus === POD_STATUS.PENDING || (eventSource && eventSource.readyState <= 1)) && (
+                <div className="flex left event-source-status">
+                    <Progressing />
+                </div>
+            )}
+        </div>
+    )
+}
+
+function useCIEventSource(url: string, maxLength?: number) {
```

Comment:
can we extract this to a different file? Something named useCIEventSource or something?

Reply:
This is required in this file only